test/integration/noderesourcetopology_test.go

```diff
@@ -0,0 +1,636 @@
+									Capacity:    intstr.FromString("2"),
+								},
+							},
+						},
+						topologyv1alpha1.Zone{
+							Name: "node-1",
+							Type: "Node",
+							Resources: topologyv1alpha1.ResourceInfoList{
+								topologyv1alpha1.ResourceInfo{
+									Name:        "foo",
+									Allocatable: intstr.FromString("2"),
+									Capacity:    intstr.FromString("2"),
+								},
+							},
+						},
+					},
+				},
+				{
+					ObjectMeta:       metav1.ObjectMeta{Name: "fake-node-2", Namespace: ns.Name},
+					TopologyPolicies: []string{"foo"},
+					Zones: topologyv1alpha1.ZoneList{
+						topologyv1alpha1.Zone{
+							Name: "node-0",
+							Type: "Node",
+							Resources: topologyv1alpha1.ResourceInfoList{
+								topologyv1alpha1.ResourceInfo{
+									Name:        "foo",
+									Allocatable: intstr.FromString("2"),
+									Capacity:    intstr.FromString("2"),
+								},
+							},
+						},
+						topologyv1alpha1.Zone{
+							Name: "node-1",
+							Type: "Node",
+							Resources: topologyv1alpha1.ResourceInfoList{
+								topologyv1alpha1.ResourceInfo{
+									Name:        "foo",
+									Allocatable: intstr.FromString("2"),
+									Capacity:    intstr.FromString("2"),
+								},
+							},
+						},
+					},
+				},
+			},
+			//both nodes are possible
+			expectedNodes: []string{"fake-node-1", "fake-node-2"},
+		},
+		//TODO: Add more test cases
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			t.Logf("Start-topology-match-test %v", tt.name)
+
+			defer cleanupNodeResourceTopologies(ctx, topologyClient, tt.nodeResourceTopologies)
+
+			if err := createNodeResourceTopologies(ctx, topologyClient, tt.nodeResourceTopologies); err != nil {
+				t.Fatal(err)
+			}
+
+			defer testutils.CleanupPods(cs, t, tt.pods)
+			// Create Pods
+			for i := range tt.pods {
+				t.Logf("Creating Pod %q", tt.pods[i].Name)
+				_, err := cs.CoreV1().Pods(ns.Name).Create(context.TODO(), tt.pods[i], metav1.CreateOptions{})
+				if err != nil {
+					t.Fatalf("Failed to create Pod %q: %v", tt.pods[i].Name, err)
+				}
+			}
+
+			for i, p := range tt.pods {
+
+				// Wait for the pod to be scheduled.
+				err = wait.Poll(1*time.Second, 20*time.Second, func() (bool, error) {
+					return podScheduled(cs, ns.Name, p.Name), nil
+				})
+				if err != nil {
+					t.Errorf("pod %q to be scheduled, error: %v", tt.pods[i].Name, err)
+				}
+
+				t.Logf(" p scheduled: %v", p.Name)
+
+				// The other pods should be scheduled on the small nodes.
+				nodeName, err := getNodeName(cs, ns.Name, p.Name)
+				if err != nil {
+					t.Log(err)
+				}
+				if contains(tt.expectedNodes, nodeName) {
+					t.Logf("Pod %q is on a nodes as expected.", p.Name)
+					continue
+				} else {
+					t.Errorf("Pod %s is expected on node %s, but found on node %s",
+						tt.pods[i].Name, tt.expectedNodes, nodeName)
+				}
+
+			}
+			t.Logf("case %v finished", tt.name)
+		})
+	}
+}
+
+func contains(s []string, e string) bool {
```

Comment:
Shouldn't we use `strings.Contains()`?

Reply:
Strings.Contains() is to check is a substring is within a string., whereas here we are checking for presence of a string in a slice of strings.

Reply:
sry, I misread.